src/Microsoft.ML.Data/Model/ModelHeader.cs

```diff
@@ -20,10 +20,13 @@ public struct ModelHeader
         public const ulong SignatureValue = 0x4C45444F4D004C4DUL;
         public const ulong TailSignatureValue = 0x4D4C004D4F44454CUL;
 
+        private const uint VerAssemblyNameSupported = 0x00010002;
+
         // These are private since they change over time. If we make them public we risk
         // another assembly containing a "copy" of their value when the other assembly
         // was compiled, which might not match the code that can load this.
-        private const uint VerWrittenCur = 0x00010001;
+        //private const uint VerWrittenCur = 0x00010001; // Initial
+        private const uint VerWrittenCur = 0x00010002; // Added AssemblyName
         private const uint VerReadableCur = 0x00010001;
```

Comment:
>0x00010001 [](start = 44, length = 10)

Hi @eerhardt, this looks mostly good thank you.

I have tested out your change locally, and noticed that when I try to load a *new* model in the *old* code, it fails with invalid format. It should fail with a more informative message that it is an invalid version, and I suspect the reason why it does not is you did not update this.

`VerReadableCur` should be bumped, if you did not intend the new format to be readable by the old software. If you *did* intend the new format to still be readable by the old software, you I have I think a bug somewhere.

Reply:
Unfortunately, there isn't room in the unused portion of the header as the assembly names are rather long strings.

There are non-repository scenarios to worry about right? I don't have access to the `Repository` in  the `TryLoadModelCore` method, where it tries to create the instance from the `ComponentCatalog`.

For now, I've bumped the `VerReadableCur` value.

Reply:
Well all right. It might have been nice to do but whatever, forwards compatibility I guess isn't a primary goal. We've just never broken it so starkly as we have here. 😄 But perhaps it's warranted.